### PR TITLE
UNF Embiggen - take four

### PR DESCRIPTION
**Pull request by Agetian**
The fourth attempt at cherry-picking the contents of the Embiggen branch after a failed merge. Hopefully fourth time is the charm and all that.

### CODE REVIEW

File: forge-game/src/main/java/forge/game/ability/AbilityUtils.java [hunk 1]
```diff
@@ -2634,6 +2634,30 @@ public static int xCount(Card c, final String s, final CardTraitBase ctb) {
...
+        if (sq[0].equals("AllTypes")) {
+            List<Card> cards = getDefinedCards(c, sq[1], ctb);
+
+            int amount = countCardTypesFromList(cards, false) +
+                    countSuperTypesFromList(cards) +
+                    countSubTypesFromList(cards);
+
+            return doXMath(amount, expr, c, ctb);
+        }
+        if (sq[0].contains("CardTypes")) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/AbilityUtils.java:2646]: still lots of bad changes in this file that already got refactored
please no changes except related for the new AllTypes property

Comment (Agetian) [reply forge-game/src/main/java/forge/game/ability/AbilityUtils.java:2646]: Which ones exactly are you referring to? I'm trying to figure out what is what here tbh and I got quite tangled in these changes :/

Comment (Agetian) [reply forge-game/src/main/java/forge/game/ability/AbilityUtils.java]: Ok, removed most of the stuff unrelated to AllTypes, but there's also a rename for get.... -> count.... - should that go as well or should it be kept as countXXX?

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/ability/AbilityUtils.java]: if it compiles now it's probably fine

Comment (Agetian) [reply forge-game/src/main/java/forge/game/ability/AbilityUtils.java]: Yes, tested the compile and the card itself, looks like it works fine. Will merge this one then. Thanks for looking at it! :+1:

**Comment by tehdiplomat** [comment]
@Agetian Oh i would have gotten back to it. Just no time yesterday.  Thanks though.

**Pull request merged**